### ISSUE TITLE
Quicksilver is writing excessive amount of data

### ISSUE THREAD

**Issue by iTechieGuy** [quicksilver/Quicksilver#2286]
In the last 8 day Quicksilver has writen 74GB of data, roughly 10 GB of data/day. As a comparison, Safari, my primary web broswer, has writen a total 1.74 GB in that same time period. That's a near 40x difference.

Is there a reason Quicksilver is writing, on average, nearly 10 GB of data of day? Is there some kind of event logging going on? Even with that, nearly 10GB/day seems excessive.

**Comment by skurfer** [comment]
I can’t find where, but another user reported something similar. No obvious cause comes to mind. I have about every plug-in there is enabled and don’t see anywhere near that sort of activity.

Do you have any idea *what* it’s writing? Are there any clues in Activity Monitor if you “Get Info” for the process and look at Open Files and Ports?

**Comment by iTechieGuy** [comment]
Rob,

Below is a list of my open files. Quicksilver has about 10 MB of files open, not including stuff in the Application directory, Frameworks, Fonts and Graphics. None of the files seem especially big. A couple were around 3-4 mb. I counted decimal points, just to make sure.

I use to be a developer and would be willing to run a debug version if you can think of some sort of trap or bottleneck and if that would help you.


BTW, Quicksilver Bytes Written has gone up by 14 GB in about the last three hours, now at 88 GB bytes written the last 8 days.

Here’s the files currently opened by Quicksilver (*** represents my user directory):

cwd
/
txt
/Applications/Quicksilver.app/Contents/MacOS/Quicksilver
txt
/Applications/Quicksilver.app/Contents/Frameworks/QSFoundation.framework/Versions/A/QSFoundation
txt
/Applications/Quicksilver.app/Contents/Frameworks/QSEffects.framework/Versions/A/QSEffects
txt
/Applications/Quicksilver.app/Contents/Frameworks/QSCore.framework/Versions/A/QSCore
txt
/Applications/Quicksilver.app/Contents/Frameworks/QSInterface.framework/Versions/A/QSInterface
txt
/usr/share/icu/icudt57l.dat
txt
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/AccessibilityAppearance.car
txt
/System/Library/Keyboard Layouts/AppleKeyboardLayouts.bundle/Contents/Resources/AppleKeyboardLayouts-L.dat
txt
/System/Library/CoreServices/Dock.app/Contents/Resources/trashempty.png
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.blacktree.Quicksilver.QSTerminalPlugIn.139.qsplugin/Contents/MacOS/Terminal Plugin
txt
/Applications/Quicksilver.app/Contents/Resources/defaultAction.icns
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.blacktree.Quicksilver.Cyberduck.39.qsplugin/Contents/MacOS/Cyberduck Plugin
txt
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/SystemAppearance.car
txt
/System/Library/CoreServices/SystemAppearance.bundle/Contents/Resources/Assets.car
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.blacktree.Quicksilver.QSSafariPlugIn.137.qsplugin/Contents/MacOS/Safari Plugin
txt
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/HIToolbox.rsrc
txt
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/English.lproj/Localized.rsrc
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.blacktree.Quicksilver.QSAppleMailPlugIn.11D.qsplugin/Contents/MacOS/Apple Mail Plugin
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.blacktree.Quicksilver.QSClipboardPlugIn.110.qsplugin/Contents/MacOS/Clipboard Plugin
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.robertson.Quicksilver.OnePassword.347.qsplugin/Contents/MacOS/OnePassword
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.robertson.Quicksilver.OnePassword.347.qsplugin/Contents/Frameworks/YAJL.framework/Versions/A/YAJL
txt
/Users/***/Library/Application Support/Quicksilver/PlugIns/com.blacktree.Quicksilver.QSServicesMenuPlugIn.DE.qsplugin/Contents/MacOS/Services Menu Plugin
txt
/Applications/Quicksilver.app/Contents/PlugIns/Core Support.qsplugin/Contents/MacOS/Core Support
txt
/Applications/Quicksilver.app/Contents/Resources/QSLigature.png
txt
/System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vImage.framework/Versions/A/Libraries/libCGInterfaces.dylib
txt
/System/Library/Components/AppleScript.component/Contents/MacOS/AppleScript
txt
/Applications/Quicksilver.app/Contents/PlugIns/Keyboard Triggers.qsplugin/Contents/MacOS/Keyboard Triggers
txt
/Applications/Quicksilver.app/Contents/PlugIns/Primer Interface.qsplugin/Contents/MacOS/Primer Interface
txt
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/Extras2.rsrc
txt
/Applications/Quicksilver.app/Contents/Resources/Button-GearMenu.tiff
txt
/System/Library/Fonts/Helvetica.dfont
txt
/Applications/Quicksilver.app/Contents/Resources/Action.png
txt
/System/Library/Fonts/SFNSText.ttf
txt
/private/var/folders/2p/s8t6_zfx4ml1knzxcsn2zhmh0000gr/0/com.apple.LaunchServices-175-v2.csstore
txt
/System/Library/PrivateFrameworks/AppleScript.framework/Versions/A/AppleScript
txt
/usr/lib/dyld
0
/dev/null
1
/dev/null
2
/dev/null
3
count=0, state=0xa
4
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/HIToolbox.rsrc
5
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/English.lproj/Localized.rsrc
6
/Applications
7
/Applications/Utilities
8
/Users/***/Library/Safari/Bookmarks.plist
9
/Users/***/Library/Safari/History.plist
10
/Users/***
11
/Users/***/Downloads
12
/Users/***/Documents
13
/Users/***/Desktop
14
/System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/Resources/Extras2.rsrc

**Comment by skurfer** [comment]
Here’s a long shot. There was a bug for a while that led to `~/Library/Application Support/Quicksilver/Triggers.plist` getting really bloated. For example, mine is 55kB now that it’s fixed, but was closer to 10MB before. This file shouldn’t get re-written unless you’re in the preferences, but it’s all I can think of.

How big is that file on your system?

**Comment by skurfer** [comment]
Additional info from a recent mailing list post…

Off the top of my head, these are the things Quicksilver would potentially write:

* Its own preference files in `~/Library/Application Support/Quicksilver`
* Clipboard history (if you’ve enabled “Keep history when restarting”)
* Catalog indexes in `~/Library/Caches/Quicksilver/Indexes`

There are some catalog entries that update frequently, but I think most of those are flagged as “can’t be indexed” so they shouldn’t be writing anything.

Maybe try keeping a Finder window open on one or more of those locations, sorted by date modified, and see if you notice anything odd?

**Comment by iTechieGuy** [comment]
Rob,

Mine what 9k and hadn’t been touched since Jan. 25 2016. Strange. I must be happy with my setup.

Kindest Regards,

Stephen Magladry
• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>

**Comment by iTechieGuy** [comment]
Rob,

My ~/Library/Application Support/Quicksilver folder is 7.1 MB in size and was last touched on Nov. 7.


My QuickSilver Indexes folder is about 7 MB with the largest file being QSPresetRunningProcesses.qsindex weighing in at 2.7 MB.


Shooting in the dark…could it be a case of write/delete write/delete write/delete write/delete… That could allow for the write sizes to get huge but file sizes stay at normal sizes.


• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>

**Comment by HenningJ** [comment]
Maybe you could try to [monitor disk use using Instruments](https://developer.apple.com/library/content/documentation/DeveloperTools/Conceptual/InstrumentsUserGuide/MeasuringIO.html#//apple_ref/doc/uid/TP40004652-CH83-SW6). I don't know if that would give you any useful infromation, but it might be worth a try.

**Comment by iTechieGuy** [comment]
I’d be more that willing to help you to hopefully find this craziness. It is the Thanksgiving weekend, so I don’t know how much time in the real near future. I’ll also have to reinstall xCode. I had to delete that awhile back to regain some disk space. Knowing this the be a short term project (hopefully ;:^) ), will lessen the blow.


• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>

**Comment by iTechieGuy** [comment]
Okay, it didn’t take as much time as I expected.

I bet matching Instrumented app to the code might be important. If it does, I'm running 1.4.1 (1419).

My hope is there some leaks along the way as opposed to periodic blowouts. Here is about a minute of file instrumentation on Quicksilver with Quicksilver running in the background. Here’s a link to the file, https://www.dropbox.com/sh/pqpurncu0xtl8ne/AABChwdKRZsn_PqY1FpAFgRba?dl=0 <https://www.dropbox.com/sh/pqpurncu0xtl8ne/AABChwdKRZsn_PqY1FpAFgRba?dl=0>.

Let me know if I can help in any other way in tracking this problem down.

• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>

**Comment by skurfer** [comment]
I’m not sure if you follow the mailing list, but a user there reports that the Chrome plug-in was to blame in his case. It doesn’t look like you’re using that, though.

I looked at the Instruments file and there weren’t many writes, but every one of them was nearly 3MB to files like `~/Library/Caches/Quicksilver/Indexes/.dat.nosync0303.Ex8ZZI`. I’m not sure what that could be from yet. (I have no such files.)

**Comment by iTechieGuy** [comment]
I’m currently only running 11 plugins. Hopefully, a little later on today, I turnoff a plugin at a time to see If I  can narrow exactly which one of my plugins are misbehaving.

I currently have:

1Password	V. 2.6.5
Apple Mail	V. 3.0.0
Automator	V. 1.2.1
Calendar		V. 1.1.2
Clipboard		V. 1.2.1
Contact		V. 2.1.0
Cyberduck	V. 2.1.0
Email		V. 2.0.0
Safari		V.2.1.5
Service Menu	V .1.3.0
Terminal		V. 2.2.2


[mostly in jest] Sure would be nice to automaticly check to see in plugins were up to date. Perhaps that could be a feature for QuickSilver 4.0. [/mostly in jest]
• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>

**Comment by tiennou** [comment]
It seems to come from the history. I'm poking at it using `opensnoop/iosnoop` (which requires SIP to be disabled) and from time to time, `iosnoop` prints a bunch of lines about a `History` file *somewhere*. More snooping (pun intended), here is one (likely differs by machine though) :
```
-rw-r--r--@ 1 tiennou  staff   109M  5 déc 20:10 /var/folders/f4/b3wl780s12gfc25kpvl1rhk40000gn/T/History
```

Relevant `iosnoop` output, for one "occurrence" (creating a Chrome tab, searching for "test"):
```
2016 Dec  5 20:17:41 ??        1   4   501  1395 W 142438816  65536 ??/Indexes/.dat.nosync0573.16AFPm Quicksilver\0
2016 Dec  5 20:17:41 ??        1   4   501  1395 W 142438944  40960 ??/Indexes/.dat.nosync0573.4atugO Quicksilver\0
2016 Dec  5 20:17:42 ??        1   4   501  1395 W 142439496  65536 ??/Indexes/.dat.nosync0573.0DTTI3 Quicksilver\0
2016 Dec  5 20:17:42 ??        1   4   501  1395 W 142440304  40960 ??/Indexes/.dat.nosync0573.XESlpt Quicksilver\0
2016 Dec  5 20:17:42 ??        1   4   501  1395 W 142443464  65536 ??/Indexes/.dat.nosync0573.jYk0ru Quicksilver\0
2016 Dec  5 20:17:43 ??        1   4   501  1395 W 161896488  65536 ??/Indexes/.dat.nosync0573.XjHy7U Quicksilver\0
2016 Dec  5 20:17:44 ??        1   4   501  1395 W 161896784  40960 ??/Indexes/.dat.nosync0573.2uK0WJ Quicksilver\0
2016 Dec  5 20:17:52 ??        1   4   501  1395 R 488223488  98304           ??/Default/History Quicksilver\0
2016 Dec  5 20:17:52 ??        1   4   501  1395 R 488223680 1048576           ??/Default/History Quicksilver\0
2016 Dec  5 20:17:52 ??        1   4   501  1395 R 488225728 1048576           ??/Default/History Quicksilver\0
2016 Dec  5 20:17:52 ??        1   4   501  1395 R 488227776  49152           ??/Default/History Quicksilver\0
2016 Dec  5 20:17:52 ??        1   4   501  1395 R 488227888 991232           ??/Default/History Quicksilver\0
2016 Dec  5 20:17:52 ??        1   4   501  1395 R 488229824 1048576           ??/Default/History Quicksilver\0
[snip about 500 lines]
2016 Dec  5 20:17:53 ??        1   4   501  1395 W 924226832 1048576                 ??/T/History Quicksilver\0
2016 Dec  5 20:17:53 ??        1   4   501  1395 W 924228880 1048576                 ??/T/History Quicksilver\0
2016 Dec  5 20:17:53 ??        1   4   501  1395 W 924230928 1048576                 ??/T/History Quicksilver\0
2016 Dec  5 20:17:53 ??        1   4   501  1395 W 924232976 1048576                 ??/T/History Quicksilver\0
2016 Dec  5 20:17:53 ??        1   4   501  1395 W 924235024 1048576                 ??/T/History Quicksilver\0
2016 Dec  5 20:17:53 ??        1   4   501  1395 W 924237072 1048576                 ??/T/History Quicksilver\0
```

**Comment by tiennou** [comment]
I have not been able to pinpoint the Default/History file. Oh wait, that's a `R`ead, and I'm pretty sure that's from the Chrome App Support dir.

The T/History one is a mystery though. Are we in some way *copying* that file in our temporary directory ??

**Comment by tiennou** [comment]
The `.dat.nosync*` files are caused by the `atomically:` flag of some write methods. I can seem them in the opensnoop output, but they're "alone" — I can't see the actual rename step, even while snooping every process and grepping on `~/Library/Caches/Quicksilver`, so I don't know which catalog entry does this...

**Comment by skurfer** [comment]
As for which catalog entry does this, I think the answer is “all of them”. (Individual entries aren’t responsible for managing the indexes on disk.)

https://github.com/quicksilver/Quicksilver/blob/master/Quicksilver/Code-QuickStepCore/QSCatalogEntry.m#L558

I wonder which one has a 3MB index and is getting frequently rewritten, though. @iTechieGuy, could you run this in Terminal?

    du -sk ~/Library/Caches/Quicksilver/Indexes/* | sort -n

The larger files will be at the bottom.

**Comment by skurfer** [comment]
> Sure would be nice to automaticly check to see in plugins were up to date.

When it checks for application updates, it also checks for new plug-ins. You won’t necessarily see the latest plug-ins, but you’ll see the latest for your version of QS and macOS. You can also manually check by looking at the “Installed Version” and “Latest Version” in the info drawer for a plug-in.

(Also, 1.4.2 is the latest official release of the main application, though I don’t think there are any plug-ins that require a newer version than 1.4.1.)

**Comment by iTechieGuy** [comment]
Sorry for the modest delay, It looks like I forgot to hit send.

Most, no, I take that back. All were behaving imho. Here’s all the indexes larger than 1/2 meg.

512	/Users/***/Library/Caches/Quicksilver/Indexes/QSPresetAddressBook.qsindex
548	/Users/***/Library/Caches/Quicksilver/Indexes/QSPresetAllApplications.qsindex
616	/Users/***/Library/Caches/Quicksilver/Indexes/QSPresetSafariHistory.qsindex
952	/Users/***/Library/Caches/Quicksilver/Indexes/QSPresetRunningProcesses.qsindex
2140	/Users/***/Library/Caches/Quicksilver/Indexes/QSPresetSafariBookmarks.qsindex



• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>

**Comment by iTechieGuy** [comment]
By no action on my parts, since I have made no changes to Quicksilver or plugins the leak that I had experienced has been plugged.  Disk writes have returned to usual and expected ranges.


BTW, I was just poking around Activity Monitor and noticed that it has sampling capabilities build it. All one has to do is select the process, click on the Gear Menu and choose Sample Process. If the results of this are similar enough to what xCode produces, someone would not have to install xCode in order to preform a sample.


Thanks for all your help, both on this problem and the project as a whole,

Stephen Magladry

• • • • • • • • • • • • • • • • • • 
Stephen Magladry
iTechieGuy.com

206 369 3606
Twitter: iTecheGuy <http://twitter.com/iTechieGuy>
Email: itechieguy@gmail.com
LinkedIn: Profile <http://www.linkedin.com/pub/stephen-magladry/29/401/71b>
